sqldelight-idea-plugin/src/main/kotlin/com/squareup/sqldelight/intellij/actions/file/SqlDelightCreateFileAction.kt

```diff
@@ -0,0 +1,186 @@
+package com.squareup.sqldelight.intellij.actions.file
+
+import com.alecstrong.sql.psi.core.psi.SqlTableName
+import com.alecstrong.sql.psi.core.psi.SqlTypes.TABLE
+import com.intellij.ide.actions.CreateFileFromTemplateAction
+import com.intellij.ide.actions.CreateFileFromTemplateDialog
+import com.intellij.ide.fileTemplates.FileTemplate
+import com.intellij.ide.fileTemplates.FileTemplateManager
+import com.intellij.ide.fileTemplates.actions.AttributesDefaults
+import com.intellij.ide.fileTemplates.ui.CreateFromTemplateDialog
+import com.intellij.openapi.actionSystem.CommonDataKeys
+import com.intellij.openapi.actionSystem.DataContext
+import com.intellij.openapi.application.runWriteAction
+import com.intellij.openapi.editor.LogicalPosition
+import com.intellij.openapi.fileEditor.FileEditorManager
+import com.intellij.openapi.project.DumbAware
+import com.intellij.openapi.project.DumbService
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.InputValidatorEx
+import com.intellij.openapi.vfs.VfsUtil
+import com.intellij.psi.PsiDirectory
+import com.intellij.psi.PsiFile
+import com.intellij.psi.util.PsiTreeUtil
+import com.intellij.testFramework.LightVirtualFile
+import com.intellij.util.IncorrectOperationException
+import com.squareup.sqldelight.core.SqlDelightPropertiesFile
+import com.squareup.sqldelight.core.lang.SqlDelightFile
+import com.squareup.sqldelight.core.lang.SqlDelightFileType
+import com.squareup.sqldelight.core.lang.SqlDelightLanguage
+import com.squareup.sqldelight.intellij.SqlDelightFileIconProvider
+import com.squareup.sqldelight.intellij.propertiesFolder
+import java.util.Properties
+
+/**
+ * Creates a new SqlDelight file/table/migration from a template (see [fileTemplates.internal])
+ *
+ * Adopted from [NewKotlinFileAction](https://github.com/JetBrains/kotlin/blob/f6a739bbc5b66a8f2cb5cb61acb1e87c899fa11e/idea/src/org/jetbrains/kotlin/idea/actions/NewKotlinFileAction.kt#L56)
+ */
+class SqlDelightCreateFileAction : CreateFileFromTemplateAction(
+    CAPTION,
+    "Creates new SqlDelight file or table",
+    SqlDelightFileType.icon
```

Comment:
I didn't really find a way how to dynamically supply the icon for the action - looks like they are being created even before the projects is loaded (`ProjectManager.getInstance().openProjects` is empty by this time). So the icon in the menu will be always static it seems.. therefore I'm not sure whether I should use a static icon for the dialog as well or use the iconprovider

Reply:
yup i think this is a reasonable default. If we want to make it unique or do something different later we can handle then